metrics/core/lib/src/data/model/build_data.dart

```diff
@@ -25,12 +27,44 @@ class BuildData extends Build implements DataModel {
           coverage: coverage,
         );
 
+  /// The unique identifier of this build.
+  String get documentId => '${projectId}_$id';
+
+  /// Creates the new instance of the [BuildData] based on current instance.
+  ///
+  /// If any of the passed parameters are null, or parameter isn't specified,
+  /// the value will be copied from the current instance.
```

Comment:
```suggestion
  /// Creates a new instance of the [BuildData] based on this one.
  ///
  /// If an argument is `null` (either `null` or nothing is passed) the value is copied 
  /// from the corresponding field of this instance.
```
?

Reply:
`this one` feels a bit javascriptey - maybe based on `current instance` would be better?
here is an example from `TextTheme`

> Creates a copy of this text theme but with the given fields replaced with the new values.

Maybe we can use something similar?

Reply:
The reason is that Effective Dart [recommends](https://dart.dev/guides/language/effective-dart/documentation#prefer-using-this-instead-of-the-to-refer-to-a-members-instance) using `this` when referencing the instance members. Maybe we can change this to `this instance`. Generally, I suppose that using `current` may be confusing making the BuildData a trackable entity that changes over time (like a user).

Reply:
While they say about `this` they are not saying about `this one` - I feel like example description from TextTheme (that includes `this`) is generally better.